packages/internal/src/callback.js

```diff
@@ -180,82 +191,124 @@ harden(isCallback);
-    }
-    methodNames = Object.keys(interfaceGuard.methodGuards);
-  }
-
-  // Cache the names for callback verification.
-  const callbackKeys = new Set(methodNames);
-
   /**
    * @typedef {(this: any, ...args: unknown[]) => any} Method
+   * @typedef {{ [ K in M ]: Method }} Methods
    */
-  /** @type {Record<any, Method>} */
-  const methods = {};
-  for (const key of callbackKeys) {
-    // Only allow the `PropertyKey` type for the target method key.
-    if (!isPropertyKey(key)) {
-      throw Fail`key ${q(key)} is not a PropertyKey`;
-    }
+  const objMethods = Object.fromEntries(
+    methodNames.map(key => {
+      // Only allow the `PropertyKey` type for the target method key.
+      if (!isPropertyKey(key)) {
+        throw Fail`key ${q(key)} is not a PropertyKey`;
+      }
 
-    methods[key] = {
-      // Explicitly use concise method syntax to preserve `this` but prevent
-      // constructor behavior.
-      /** @type {Method} */
-      [key](...args) {
-        // Support both synchronous and async callbacks.
-        const cb = this.state.cbs[key];
-        cb || Fail`unimplemented method ${q(key)}`;
-        if (cb.isSync) {
-          return callSync(cb, ...args);
-        }
-        return callE(cb, ...args);
-      },
-    }[key];
-  }
+      const m = /** @type {Methods} */ ({
+        // Explicitly use concise method syntax to preserve `this` but prevent
+        // constructor behavior.
+        /** @type {Method} */
+        [key](...args) {
+          // Support both synchronous and async callbacks.
+          const cb = this.state.cbs[key];
+          if (!cb) {
+            const err = assert.error(
+              `unimplemented ${q(tag)} method ${q(key)}`,
+            );
+            if (this.state.isSync) {
+              throw err;
+            }
+            return Promise.reject(err);
+          }
+          if (cb.isSync) {
+            return callSync(cb, ...args);
+          }
+          return callE(cb, ...args);
+        },
+      })[key];
+      return /** @type {const} */ ([key, m]);
+    }),
+  );
+  const methods = /** @type {Methods} */ (objMethods);
+
+  const methodKeys = getOwnPropertyKeys(methods);
 
   /**
-   * Create an exo object whose behavior is composed from method names mapped
-   * to individual callbacks.  Nullish callbacks are ignored.
+   * Create an exo object whose behavior is composed from a default target
+   * and/or individual method override callbacks.
    *
-   * @type {<Cbs extends Record<any, Callback<any> | null>>(callbacks: Cbs) => MayFar<Record<keyof Cbs, (...args: unknown[]) => any)>}
+   * @param {object} opts
+   * @param {unknown} [opts.target] The target for any methods that
+   * weren't specified in `opts.overrides`.
+   * @param {boolean} [opts.isSync=false] Whether the target should be treated
+   * as synchronously available.
+   * @param {Record<M, Callback<any> | null>} [opts.overrides={}] Set individual
```

Comment:
I think the intent is to have the override be optional, which `Record` does not express.

```suggestion
   * @param {{[key: M]?: Callback<any> | null}} [opts.overrides={}] Set individual
```

Reply:
I had to use a mapped object type instead of an index type, but that worked!